tests/src/serverTestArrayPersistence.cpp

```diff
@@ -0,0 +1,140 @@
+#define BOOST_TEST_MODULE serverTestArrayPersistence
+#include <boost/test/included/unit_test.hpp>
+
+#include "DoocsAdapter.h"
+#include "serverBasedTestTools.h"
+#include <ChimeraTK/ControlSystemAdapter/Testing/ReferenceTestApplication.h>
+#include <doocs-server-test-helper/doocsServerTestHelper.h>
+
+extern const char* object_name;
+#include <doocs-server-test-helper/ThreadedDoocsServer.h>
+
+#include <vector>
+#include <fstream>
+
+using namespace boost::unit_test_framework;
+using namespace boost::unit_test;
+using namespace ChimeraTK;
+
+template<typename T>
+void vectorToFile(std::vector<T>& vec, const char* fname) {
+  std::ofstream of(fname);
+  for(auto num : vec) of << num << std::endl;
+  of.close();
+}
+template<typename T>
+void vectorFromFile(std::vector<T>& vec, const char* fname) {
+  std::ifstream ifs(fname);
+  size_t j = 0;
+  T num;
+  while((ifs >> num) && j < vec.size()) {
+    vec[j++] = num;
+  }
+  ifs.close();
+}
+
+const int alen = 40;
+
+// slightly modified GlobalFixture from serverBasedTestTools.h
```

Comment:
please explain in the comment in which way it has been modified and why it is not (easily) possible to integrate it as a new feature in the common GlobalFixture.

Reply:
fixed

Reply:
Could be a bit more elaborate, but ok.